### DISCUSSION TITLE
Could the images be loaded locally?

### DISCUSSION THREAD

**Discussion by emanuellopess** [mercadopago/sdk-js#212] [Ideas]
Could this images be included in the Mercado Pago SDK instead to be offloaded to this server?

..pago-button::before {\n background-image: url("http://static.mlstatic.com/org-img/mercadopago/wallet_mp_icon.svg");\n background-size: ${34 / 14}em ${34 / 14}em;\n width: ${34 / 14}em;\n h... ...pago-button::before {\n background-image: url("http://static.mlstatic.com/org-img/mercadopago/wallet_mp_icon.svg");\n background-size: 32px;\n backgroud-color: #fff;\n background-repeat: n...

**Comment by AleFossati** [comment]
Hello @emanuellopess, for now the short answer would be no.

The Mercado Pago SDK includes various features, and most of them need specific assets to work properly. That's why static files such as images, JS, and configurations are loaded on demand, only when you instantiate the product you are going to use.

Imagine downloading tons of Bricks files, for example, when the only thing you are using on your website is the getIdentificationTypes function from CoreMethods. It would not be a good experience.

However, we are curious to understand why you suggest these images be included directly in the SDK. Could you elaborate?

**Reply by emanuellopess** [reply]
@AleFossati This feature is required by wordpress plugin guidelines for approval in case we use the Mercado Pago Checkout Pro payment method, we received this message requiring the images to be included in the SDK:

## Calling files remotely

Offloading images, js, css, and other scripts to your servers or any remote service (like Google, MaxCDN, jQuery.com etc) is disallowed. When you call remote data you introduce an unnecessary dependency on another site. If the file you're calling isn't a part of WordPress Core, then you should include it -locally- in your plugin, not remotely. If the file IS included in WordPress core, please call that instead.

An exception to this rule is if your plugin is performing a service. We will permit this on a case by case basis. Since this can be confusing we have some examples of what are not permitted:

- Offloading jquery CSS files to Google - You should include the CSS in your plugin.
- Inserting an iframe with a help doc - A link, or including the docs in your plugin is preferred.
- Calling images from your own domain - They should be included in your plugin.

Here are some examples of what we would permit:

- Calling font families from Google or their approved CDN (if GPL compatible)
- API calls back to your server to process possible spam comments (like Akismet)
- Offloading comments to your own servers (like Disqus)
- oEmbed calls to a service provider (like Twitter or YouTube)

Please remove external dependencies from your plugin and, if possible, include all files within the plugin (that is not called remotely). If instead you feel you are providing a service, please re-write your readme.txt in a manner that explains the service, the servers being called, and if any account is needed to connect.

Example(s) from your plugin:

```
[...]/MercadoPago.js:5 ...pago-button::before {\n background-image: url("http://static.mlstatic.com/org-img/mercadopago/wallet_mp_icon.svg");\n background-size: ${34 / 14}em ${34 / 14}em;\n width: ${34 / 14}em;\n h... ...pago-button::before {\n background-image: url("http://static.mlstatic.com/org-img/mercadopago/wallet_mp_icon.svg");\n background-size: 32px;\n backgroud-color: #fff;\n background-repeat: n...
```

**Reply by AleFossati** [reply]
If I got it right, you are developing a wordpress plugin that will use our SDK, is that correct? 🤔 

From my understanding of the mentioned reference, it's not allowed to offload assets (images, static files, etc) in general **unless** your plugin provides a service, which is exactly what you are trying to accomplish (_please, let me know if I'm wrong_ 😅 ).

If a plugin is not self contained and thus use any other service, it is very likely that they will offload lots of content, since putting everything inside the first file generates a slow and degraded experience for users and for that reason is considered a bad practice.

I'm not a wordpress expert, but based on that reference that you sent, I think that the solution would be to change the `readme.txt` in order to explain what your plugin does and what the mercadopago sdk is downloading (which are assets for our payment service), as mentioned here:

> "... _If instead you feel you are providing a service, please re-write your readme.txt in a manner that explains the service, the servers being called, and if any account is needed to connect._"

If you need help with anything, please let me know.
